functions/dialect.go

```diff
@@ -176,18 +188,22 @@ func (d *dialectImpl) buildFunctionInfoMap(functions []dialectFunction) map[exte
 }
 
 type dialectFile struct {
-	Name               string                     `yaml:"name"`
-	Type               string                     `yaml:"type"`
-	SupportedTypes     map[string]dialectTypeInfo `yaml:"supported_types,omitempty"`
-	ScalarFunctions    []dialectFunction          `yaml:"scalar_functions,omitempty"`
-	AggregateFunctions []dialectFunction          `yaml:"aggregate_functions,omitempty"`
-	WindowFunctions    []dialectFunction          `yaml:"window_functions,omitempty"`
-	Dependencies       map[string]string          `yaml:"dependencies,omitempty"`
+	Name               string                      `yaml:"name"`
+	Type               string                      `yaml:"type"`
+	SupportedTypes     map[string]*dialectTypeInfo `yaml:"supported_types,omitempty"`
+	ScalarFunctions    []dialectFunction           `yaml:"scalar_functions,omitempty"`
+	AggregateFunctions []dialectFunction           `yaml:"aggregate_functions,omitempty"`
+	WindowFunctions    []dialectFunction           `yaml:"window_functions,omitempty"`
+	Dependencies       map[string]string           `yaml:"dependencies,omitempty"`
 }
 
 type dialectTypeInfo struct {
 	SqlTypeName       string `yaml:"sql_type_name"`
-	SupportedAsColumn bool   `yaml:"supported_as_column"`
+	SupportedAsColumn *bool  `yaml:"supported_as_column" default:"true"`
```

Comment:
let's avoid change to pointer

Reply:
This is needed for SetDefaults to differentiate the zero value from the provided value.

Without this it is incorrectly setting the value to true even if the input is
`supported_as_column: false`

Reply:
I tried UnmarshalYAML, one hiccup with this is we cannot use `defaults.Set`  on any struct using dialectTypeInfo. I had to remove `defaults.Set` on dialectFile. 

Looks like whenever we have `default:"true"` the choice is between 1. using pointer vs 2. adding UnmarshalYAML in all the structs (that includes this bool indirectly)  to set the defaults.

LMK 1 or 2?